bukkit-helper-121-6/src/main/java/org/dynmap/bukkit/helper/v121_6/BukkitVersionHelperSpigot121_6.java

```diff
@@ -0,0 +1,450 @@
+		}
+	}
+	
+	/**
+	 * Get material map by block ID
+	 */
+	@Override
+	public BukkitMaterial[] getMaterialList() {
+		return new BukkitMaterial[4096]; // Not used
+	}
+
+	@Override
+	public void unloadChunkNoSave(World w, org.bukkit.Chunk c, int cx, int cz) {
+		Log.severe("unloadChunkNoSave not implemented");
+	}
+
+	private String[] biomenames;
+	@Override
+	public String[] getBiomeNames() {
+		if (biomenames == null) {
+			biomenames = new String[256];
+			Iterator<BiomeBase> iter = getBiomeReg().iterator();
+			while (iter.hasNext()) {
+				BiomeBase b = iter.next();
+				int bidx = getBiomeReg().a(b);
+				if (bidx >= biomenames.length) {
+					biomenames = Arrays.copyOf(biomenames, bidx + biomenames.length);
+				}
+				biomenames[bidx] = b.toString();
+			}
+		}
+		return biomenames;
+	}
+
+	@Override
+	public String getStateStringByCombinedId(int blkid, int meta) {
+		Log.severe("getStateStringByCombinedId not implemented");
+		return null;
+	}
+	@Override
+	/** Get ID string from biomebase */
+	public String getBiomeBaseIDString(Object bb) {
+		return getBiomeReg().b((BiomeBase)bb).a(); // MinecraftKey.getPath()
+	}
+	@Override
+	public String getBiomeBaseResourceLocsation(Object bb) {
+		return getBiomeReg().b((BiomeBase)bb).toString();
+	}
+
+	@Override
+	public Object getUnloadQueue(World world) {
+		Log.warning("getUnloadQueue not implemented yet");
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public boolean isInUnloadQueue(Object unloadqueue, int x, int z) {
+		Log.warning("isInUnloadQueue not implemented yet");
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	@Override
+	public Object[] getBiomeBaseFromSnapshot(ChunkSnapshot css) {
+		Log.warning("getBiomeBaseFromSnapshot not implemented yet");
+		// TODO Auto-generated method stub
+		return new Object[256];
+	}
+
+	@Override
+	public long getInhabitedTicks(Chunk c) {
+		return ((CraftChunk)c).getHandle(ChunkStatus.n).w(); // ChunkStatus.FULL ; IChunkAccess.getInhabitedTime
+	}
+
+	@Override
+	public Map<?, ?> getTileEntitiesForChunk(Chunk c) {
+		return ((CraftChunk)c).getHandle(ChunkStatus.n).k; // ChunkStatus.FULL ; IChunkAccess.blockEntities
+	}
+
+	@Override
+	public int getTileEntityX(Object te) {
+		TileEntity tileent = (TileEntity) te;
+		return tileent.aA_().u(); // TileEntity.getBlockPos ; Vec3i.getX
+	}
+
+	@Override
+	public int getTileEntityY(Object te) {
+		TileEntity tileent = (TileEntity) te;
+		return tileent.aA_().v(); // TileEntity.getBlockPos ; Vec3i.getY
+	}
+
+	@Override
+	public int getTileEntityZ(Object te) {
+		TileEntity tileent = (TileEntity) te;
+		return tileent.aA_().w(); // TileEntity.getBlockPos ; Vec3i.getZ
+	}
+
+	@Override
+	public Object readTileEntityNBT(Object te, org.bukkit.World w) {
+		TileEntity tileent = (TileEntity) te;
+		CraftWorld cw = (CraftWorld) w;
+		return tileent.a(cw.getHandle().K_()); // TileEntity.saveCustomOnly ; LevelReader.registryAccess
```

Comment:
[Should be](https://mappings.dev/1.21.7/net/minecraft/world/level/block/entity/BlockEntity.html) tileent.d (saveCustomOnly) instead of tileent.a (getUpdateTag)